slrealizer/realize_om10.py

```diff
@@ -0,0 +1,125 @@
+        
+        galsimInput={'flux': flux,
+                       'half_light_radius': hlr,
+                       'e': e,
+                       'beta': beta,
+                       'num_objects': lensInfo['NIMG']}
+        
+        for obj in xrange(lensInfo['NIMG']):
+            obj_mag = lensInfo[band + '_SDSS_quasar']\
+                      + from_flux_to_mag(abs(lensInfo['MAG'][obj]))
+                      #+ from_flux_to_mag(lensInfo['MAG'][obj] + get_filter_AB_offset())
+            galsimInput['flux_'+str(obj)] = from_mag_to_flux(obj_mag, to_unit='nMgy') # don't subtract AB offset?
+            galsimInput['xy_'+str(obj)] = lensInfo['XIMG'][obj], lensInfo['YIMG'][obj]
+            
+        return galsimInput        
+            
+    def draw_system(self, obsInfo, lensInfo, save_dir=None):
+        galsimInput = self._from_om10_to_galsim(lensInfo, obsInfo['filter'])
+        return self.as_super.draw_system(galsimInput=galsimInput, obsInfo=obsInfo, save_dir=save_dir)
+    
+    def estimate_hsm(self, obsInfo, lensInfo):
+        galsim_img = self.draw_system(lensInfo=lensInfo, obsInfo=obsInfo, save_dir=None)
+        return self.as_super.estimate_hsm(galsim_img=galsim_img, obsInfo=obsInfo)
+    
+    def draw_emulated_system(self, obsInfo, lensInfo):
+        hsmOutput = self.estimate_hsm(obsInfo, lensInfo)
+        return self.as_super.draw_emulated_system(hsmOutput)
+    
+    def create_source_row(self, obsInfo, lensInfo):
+        hsmOutput = self.estimate_hsm(obsInfo, lensInfo)
+        if hsmOutput is None:
+            return None
+        objectId = lensInfo['LENSID']
+        return self.as_super.create_source_row(hsmOutput=hsmOutput, objectId=objectId, obsInfo=obsInfo)
+    
+    def make_source_table(self, save_dir):
+        return self.as_super.make_source_table(save_dir=save_dir)
+    
+    def draw_lens_random_date(self, lensID=None, rownum=None, save_dir=None):
+        """                                                                                                                   
+        Draws two images of the lens system with the given rownum
+        under randomly chosen observation conditions,
+        one from the catalog info (truth image) and
+        another from HSM's estimation of the truth image
+
+        Keyword arguments:
+        - lensID: an integer specifying the ID of the lens system in the OM10 catalog
+
+        Returns:
+        A tuple of
+        - the truth image drawn from the catalog info
+        - the emulated image drawn from HSM's derived info of aggregate system 
+
+        """
+        # Randomly select observation ID
+        obs_rownum = random.randint(0, self.num_obs)
+        # Render an image of the system using GalSim
+        truth_img, obj = self.draw_system(self.get_lensInfo(lensID=lensID, rownum=rownum),\
+                                    self.get_observation(rownum=obs_rownum),\
+                                    save_dir)
```

Comment:
Better function title? Method seems to be visualizing as well as calculating.

Reply:
Changed to `compare_truth_vs_emulated` (too long?)